File: leetcode442.cpp
class Solution {
public:
    vector<int> findDuplicates(vector<int>& nums) {
      vector<int>result;
      for(int i = 0;i<nums.size();i++){
        int n = abs(nums[i]);
        if(nums[n - 1] < 0){
          result.push_back(n);
        }else{
          nums[n - 1] *= -1;
        }
      }
      return result;
    }
};

File: leetcode442.java
class Solution {
    public List<Integer> findDuplicates(int[] nums) {
      HashMap<Integer,Integer>map = new HashMap<>();
      List<Integer>list = new ArrayList<>();
      for(int i = 0;i<nums.length;i++){
        if(!map.containsKey(nums[i])){
          map.put(nums[i],0);
        }else{
          list.add(nums[i]);
        }
      }
      return list;
    }
}
